pkg/epp/util/env/env.go

```diff
@@ -1,61 +1,84 @@
 package env
 
 import (
+	"fmt"
 	"os"
 	"strconv"
+	"time"
 
 	"github.com/go-logr/logr"
 )
 
-// getEnvFloat gets a float64 from an environment variable with a default value
-func GetEnvFloat(key string, defaultVal float64, logger logr.Logger) float64 {
-	val, exists := os.LookupEnv(key)
-	if !exists {
-		logger.Info("Environment variable not set, using default value",
-			"key", key, "defaultValue", defaultVal)
-		return defaultVal
-	}
-
-	floatVal, err := strconv.ParseFloat(val, 64)
+// parseEnvWithValue attempts to parse a string value using the provided
+// parser.
+// It logs success or failure and returns the parsed value or the default
+// value.
+// This helper is used when the environment variable is confirmed to exist.
+func parseEnvWithValue[T any](
+	key string,
+	valueStr string,
+	defaultVal T,
+	parser func(string) (T, error),
+	logger logr.Logger,
+	typeName string,
```

Comment:
what do you think about using `reflect.TypeOf(defaultVal)` and removing the typeName arg?
reflection comes with a cost, but since all env vars are read on bootstrap I don't think that is an issue.

Reply:
That's a great idea. The main reason I added this helper was to standardize the logs and make adding new types slightly easier. This changes supports both of those goals.

Reply:
Incorporated--thanks!